### DISCUSSION TITLE
"Submit" button doesn't do anything

### DISCUSSION THREAD

**Discussion by imranh101** [KRTirtho/spotube#61] [Q&A]
Hey, installed the app via Flatpak on my Steam Deck! If I can get this working you might see an uptick in users using this app :D I generated a webapp, added the callback URL as described, and then pasted the client ID and Secret over to the app. When clicking Submit, nothing happens. No error, no popup.

**Comment by imranh101** [comment]
Tried installing on my PC and I see that it is supposed to pop open a webbrowser after you hit submit to approve the login. I assume I am missing some sort of permission for the flatpak

**Reply by imranh101** [reply]
Tried installing Flatseal to give the app some permissions and still not able to get the browser popup to confirm login, hmmm

**Comment by KRTirtho** [comment]
I haven't tested the app on the Steam Deck yet. Did you try using the Steam Desktop mode & then using Spotube from the desktop? I guess the deck doesn't allow IPC in the Steam OS mode

**Reply by imranh101** [reply]
Yeah, that is how I was trying. The app launches and everything but when putting in the login info and hit submit it doesn't seem to pop up a browser to "Authorize an App" (or whatever it specifically does when you hit the Submit button, don't remember the exact wording of the page). I think the app doesn't have permissions to launch other applications, though I could be wrong. I tried giving if a couple of various permissions from flatseal including full filesystem perms and still no go. Thanks for the reply sir!

**Comment by KRTirtho** [comment]
You can try out the [AppImage](https://github.com/KRTirtho/spotube/releases/download/v2.0.0/Spotube-linux-x86_64.AppImage) version of Spotube. It should work fine mostly as native. Here's a guide how to integrate the AppImage with SteamDeck UI https://www.reddit.com/r/SteamDeck/comments/t9xwte/how_to_automatically_integrate_appimage_apps_into/

**Reply by imranh101** [reply]
Hey, Im sorry Ive been super busy lately and couldnt get around to it. Finally got around to grabbing appimage version and surprisingly still having the exact same issue, clicking Submit just isnt doing anything :(
